rflx/generator/session.py

```diff
@@ -530,13 +530,19 @@ def _create_declarations(
                     for t in self._session_context.used_types
                     if not model.is_builtin_type(t) and not model.is_internal_type(t)
                 ],
-                EnumerationType(
-                    "Channel",
-                    {
-                        ID(f"C_{parameter.identifier}"): None
-                        for parameter in session.parameters.values()
-                        if isinstance(parameter, decl.ChannelDeclaration)
-                    },
+                *(
+                    [
+                        EnumerationType(
+                            "Channel",
+                            {
+                                ID(f"C_{parameter.identifier}"): None
+                                for parameter in session.parameters.values()
+                                if isinstance(parameter, decl.ChannelDeclaration)
+                            },
+                        )
+                    ]
+                    if session.parameters.values()
+                    else []
```

Comment:
Channels are not the only parameters of a session, so a more precise condition is needed to always only create the enum, if there are no channels.

Reply:
OK, I used a local variable to capture the list of channel parameters.